### ISSUE TITLE
Add accessibility descriptions to entries in list of stations

### ISSUE THREAD

**Issue by y20k** [y20k/transistor#190]
This issue came in via email: Add accessibility descriptions to entries in list of stations so that screenreaders can process them.

**Comment by Moskytto** [comment]
Hello! I'm here for the first time. It was I who made this request.

**Comment by y20k** [comment]
Nice. Welcome! It would be great, if you could help me testing this. And: Please excuse me in advance. This project is moving rather slow at the moment. I only work on it in my spare time.

**Comment by y20k** [comment]
Hi @Moskytto ,

I made some minor changes to the app that in theory should fix this issue. Over the weekend I fired up Talkback for the first time. And as far as I could judge Transistor now reads out load the station names.

Would you like to test that for me? Here I have an APK file, so that you can sideload the app, if you like.

* [transistor-app-test-3.0.10-01.apk](https://www.dropbox.com/s/zizay6h9ntprppr/transistor-app-test-3.0.10-01.apk?dl=0)

We can use the opportunity to streamline some of the texts the app exposes to accessibility services, like Talkback. Feel free to suggest changes. I cannot promise to implement them all, but I will try, if it is not too much work.

**Comment by Moskytto** [comment]
Hello! Thank you very much for your attention. I was very excited to test a new version of the application that you kindly sent me. But unfortunately my device did not work. I could not test with Talk Back because the same does not open. Displays a message "The transistor has stopped". My device is Moto E 4 Plus running Android 7.1.1. Do not worry, you do not have to hurry. Keep doing this work in your spare time. Always for pleasure, not for obligation, because everything will work out. Thank you!
Translated from Portuguese, using Google Translate.

**Comment by y20k** [comment]
Hi there.
I will look into it and post a new APK. Sorry.

**Comment by Moskytto** [comment]
no problem

**Comment by y20k** [comment]
Hi,

I was able to reproduce the problem. There seemed to be a difference between the debug version I was using to test the app and the version I compiled for you. Anyway. After a few changes I managed to compile a version that should work for your device. I just tested it on my phone, too.

* [transistor-app-test-3.0.10-02.apk](https://www.dropbox.com/s/6lzhuo27u42fj8f/transistor-app-test-3.0.10-02.apk?dl=0)

PS. I did not mention it earlier. When installing test versions it is advisable to backup your radio stations. They are stored in the following folder on your external storage. `/Android/data/org.y20k.transistor/files/Collection`. Just in case the Auto-Backup feature that the OS provides, does not work.

**Comment by Moskytto** [comment]
HI! It should work, and it worked. Now I can hear the names of the stations. I have reviewed almost all the menus and buttons and the only thing that is still not read is the transmission link that is in the details of the player. But in fact this is not relevant as there is already a button to copy. I'm just reporting this detail. Now the app is perfect. Thank you so much for being so thoughtful! If I can help you with something, do not hesitate to ask.

**Comment by y20k** [comment]
Hi there,

the stream address now should also be readable. I added it as a "description" that TalkBack should read out loud. Here is a new APK:

* [transistor-app-test-3.0.10-03.apk](https://www.dropbox.com/s/ruegjdisi7008l7/transistor-app-test-3.0.10-03.apk?dl=0)

I will post the current version within the next couple of days to the F-Droid store and to the PlayStore.

**Comment by Moskytto** [comment]
Perfect! Accessibility 100%. I'll be waiting for availability in app stores. PS: I found this app on FDroid.

**Comment by y20k** [comment]
[v3.0.10](https://github.com/y20k/transistor/releases/tag/v3.0.10) is out. It may take a couple of days until it shows up on F-Droid. This version contains the accessibility improvements. Thank you for helping me with this issue!!
